apollo/lib/websocket/app.py

```diff
@@ -0,0 +1,55 @@
+import enum
+import json
+
+from fastapi import WebSocket
+from starlette.websockets import WebSocketDisconnect
+
+from apollo.lib.decorators import with_db_session
+
+
+class WebSocketObserverInterestTypes(enum.Enum):
+    @staticmethod
+    @with_db_session
+    def _list_all_agents(session):
+        from apollo.models.agent import list_all_agents
+        from apollo.lib.schemas.agent import BaseAgentSchema
+
+        return [json.loads(BaseAgentSchema.from_orm(agent).json()) for
```

Comment:
Wouldn't it be possible to just leave it as json and use `send_text` *or* use `dict()` here? So that you don't have to convert to json and back to a dict.

Reply:
If i remove the json.loads, the output is:

 `["{\"id\": \"4a26ed7d-3002-40fc-82e2-4ed6f88763f8\", \"name\": \"test\", \"connection_state\": \"disconnected\"}"]`

Then the receiver must `json.load` every list item individually because list of strings.

Reply:
So then the `dict()` route would be appropriate.

```
from pydantic.json import pydantic_encoder

[BaseAgentSchema.from_orm(agent).dict() for ...]

# later
...
connection.send_text(json.dumps(data, default=pydantic_encoder))
```

Unless you can give the encoder to `send_json` but it isn't documented in Starlette. You'd think there'd be (de)serialisation support for multiple entities but don't see that documented in Pydantic either.